### DISCUSSION TITLE
duoLayer but layer should be activated with only one of the two keys

### DISCUSSION THREAD

**Discussion by kaygdotorg** [evan-liu/karabiner.ts#145] [Q&A] [answered]
A common problem I have with duoLayer (which is why I use hyperLayer) instead is that it affects my typing speed. I can type 80 wpm on a slow day, 120 on a better day so with many duoLayers in my configuration, there's a slight delay in typing alphabets because I assume karabiner waits to see if i press the other duolayer key next to activate it. 

For example: 

a + ; → Navigation Layer
s + ; → Numpad Layer
d + ; → Raycast Layer
f + ; → Symbol Layer

Now whenever I come across words that have any of those alphabets in, it's noticeably slower to type. 

So I was looking for a way to have:

`a + ;` → Navigation Layer 

BUT the layer is only triggered when `;` is pressed FIRST, then `a`. 

Is something like this possible?

Thank you!

**Comment by evan-liu** [comment, marked as answer]
There are 2 things you can try if help: 

```ts
duoLayer(';', 'a')
    .threshold(30) // Can also be global 'duo_layer.threshold_milliseconds'
    .options({ key_down_order: 'strict' })

```

**Reply by kaygdotorg** [reply]
Let me try that and report back. Thank you so much!

**Reply by kaygdotorg** [reply]
Hey, `key_down_order` works really well. I am sure the threshold modifier would work just as well if I run into problems still.
